src/components/ha-push-notifications-toggle.ts

```diff
@@ -12,42 +19,32 @@ export const pushSupported =
     document.location.hostname === "localhost" ||
     document.location.hostname === "127.0.0.1");
 
-/*
- * @appliesMixin EventsMixin
- */
-class HaPushNotificationsToggle extends EventsMixin(PolymerElement) {
-  static get template() {
+@customElement("ha-push-notifications-toggle")
+class HaPushNotificationsToggle extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public disabled = false;
+
+  @property() private _loading = true;
+
+  @property() private _checked = false;
+
+  protected render(): TemplateResult {
     return html`
       <ha-switch
-        disabled="[[_compDisabled(disabled, loading)]]"
-        checked="{{pushChecked}}"
-        on-change="handlePushChange"
+        .disabled2=${this.disabled || this._loading}
+        .checked=${this._checked}
+        @change=${this._handlePushChange}
       ></ha-switch>
     `;
   }
 
-  static get properties() {
-    return {
-      hass: { type: Object, value: null },
-      disabled: {
-        type: Boolean,
-        value: false,
-      },
-      pushChecked: {
-        type: Boolean,
-        value:
-          "Notification" in window && Notification.permission === "granted",
-      },
-      loading: {
-        type: Boolean,
-        value: true,
-      },
-    };
-  }
-
-  async connectedCallback() {
+  public async connectedCallback() {
```

Comment:
Not async

Reply:
Not even close to ready, hence the draft and WIP status :)

Reply:
I know :-)